### PR TITLE
Prevent users from leaving form views with unsaved changes

### PR DESCRIPTION
**Pull request by noahlocke**
Requires laravel/nova#481

@jbrooksuk suggested I make this a mixin, since it will be duplicated across several views.

### CODE REVIEW

File: src/mixins/BeforeRouteLeave.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+export default {
+    beforeRouteLeave(to, from, next) {
+        if (this.canLeave) {
+            next()
+            return;
+        }
+
+        const answer = window.confirm(
+            this.__('Do you really want to leave? You have unsaved changes!')
```
Comment (jbrooksuk) [inline src/mixins/BeforeRouteLeave.js]: > Do you really want to leave? You have unsaved changes.

File: src/mixins/BeforeRouteLeave.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        )
+        if (answer) {
+            next()
+        } else {
```
Comment (jbrooksuk) [inline src/mixins/BeforeRouteLeave.js]: If we return after the call to `next()` above, we can drop this else.

File: src/mixins/BeforeRouteLeave.js (hunk 1 above)
Comment (noahlocke) [reply src/mixins/BeforeRouteLeave.js]: Fixed in https://github.com/noahlocke/nova-js/commit/1757ba07a13bb80de95d29eddc545f01ee8e7c75

File: src/mixins/BeforeRouteLeave.js (hunk 2 above)
Comment (noahlocke) [reply src/mixins/BeforeRouteLeave.js]: Fixed in https://github.com/noahlocke/nova-js/commit/1757ba07a13bb80de95d29eddc545f01ee8e7c75

**Comment by davidhemphill** [comment]
We should rename this to better communicate what it does instead of where it goes. :-)

**Comment by noahlocke** [comment]
@davidhemphill done, does this work?

**Comment by jbrooksuk** [comment]
@noahlocke I think David was referring to the function name :)

**Comment by noahlocke** [comment]
@jbrooksuk OK, sorry, now I'm confused. `beforeRouteLeave()` is a Vue function, or are you just referring to the mixin itself? Rename to something more like `PreventFormAbandon.js`?

**Comment by davidhemphill** [comment]
Yes the mixin name itself. :-)

**Comment by noahlocke** [comment]
@davidhemphill @jbrooksuk done.

**Comment by JavierMartinz** [comment]
What's the status of this?

**Comment by davidhemphill** [comment]
The parent PR is closed.

**Pull request closed without merging**